src/state.rs

```diff
@@ -30,14 +38,52 @@ pub struct State {
 }
 
 impl State {
+    /// Loads the state.json from the filesystem, waiting for a file lock to ensure no other
+    /// concurrent Kubie processes are accessing/writing the file at the same time.
     pub fn load() -> Result<State> {
+        Self::access(|state| Ok(state))
+    }
+
+    /// Takes a closure that allows for modifications of the state. Automatically handles
+    /// locking/unlocking and saving after execution of the closure.
+    pub fn modify<F: FnOnce(&mut State) -> Result<()>>(func: F) -> Result<()> {
+        Self::access(|mut state| {
+            func(&mut state)?;
+            state.save()?;
+            Ok(())
+        })
+    }
+
+    fn access<R, F: FnOnce(State) -> Result<R>>(func: F) -> Result<R> {
+        let path = KUBIE_STATE_LOCK_PATH.display();
+
+        // Acquire the lock
+        let flock = File::create(lock_path())?;
```

Comment:
I suspect this is where the bug in #69 is coming from, since the directory that this gets created in is probably only created in `ioutil`. I'll add some `.context()` here and try a fresh install (no pre-existing directories) to see if that's correct.

Reply:
Ah right of course. It's missing the `mkdir -p` with DirBuilder. I was trying to isolate the bug as well, I can try to reproduce this.

Reply:
Yep that's it. I removed `~/.local/share/kubie` and I get the bug.

Reply:
Are you working on a fix? If you're busy I can do it.